--- a/node_modules/electron/index.js
+++ b/node_modules/electron/index.js
@@ -4,6 +4,7 @@ var path = require('path')
 var pathFile = path.join(__dirname, 'path.txt')
 
 function getElectronPath () {
+  return path.join("/usr/local/share/electron11/electron")
   if (fs.existsSync(pathFile)) {
     var executablePath = fs.readFileSync(pathFile, 'utf-8')
     if (process.env.ELECTRON_OVERRIDE_DIST_PATH) {
