--- node_modules/electron-builder/out/cli/install-app-deps.js
+++ node_modules/electron-builder/out/cli/install-app-deps.js
@@ -122,7 +122,7 @@
   return yargs.parserConfiguration({
     "camel-case-expansion": false
   }).option("platform", {
-    choices: ["linux", "darwin", "win32"],
+    choices: ["freebsd","linux", "darwin", "win32"],
     default: process.platform,
     description: "The target platform"
   }).option("arch", {

